lib/awful/keygrabber.lua

```diff
@@ -45,7 +207,411 @@ function keygrabber.stop(g)
     end
 end
 
----
+--- The keygrabber timeout.
+--
+-- @DOC_text_awful_keygrabber_timeout_EXAMPLE@
+--
+-- @property timeout
+-- @see gears.timer
+-- @see timeout_callback
+
+--- The key on which the keygrabber listen to terminate itself.
+--
+-- When this is set, the running keygrabber when quit when [one of] the release
+-- key event occurs.
+--
+-- By default, the event is `press`. It is common for use case like the
+-- `awful.prompt` where `return` (enter) will terminate the keygrabbing. Using
+-- `release` as an event is more appropriate when the keygrabber is tied to a
+-- modifier key. For example, an Alt+Tab implementation stops when `mod1` (Alt)
+-- is released.
+--
+-- It can also be a table containing many keys (as values).
+--
+-- @DOC_text_awful_keygrabber_release_keys_EXAMPLE@
+--
+-- @DOC_text_awful_keygrabber_release_key_EXAMPLE@
+--
+-- Please note that modkeys are accepted as release keys as long as the
+-- keyboard layouts use the IBM PC 104-key naming convention for the key names.
+--
+-- @property release_key
+-- @param[opt=nil] string|table release_key
+-- @see release_event
+
+--- The event on which the keygrabbing will be terminated.
+--
+-- the valid values are:
+--
+-- * "press" (default)
+-- * "release"
+--
+-- @property release_event
+-- @param string
+-- @see release_key
+
+--- Whether or not to execute the key press/release callbacks when keybindings are called.
+--
+-- When this property is set to false, the `keyreleased_callback` and
+-- `keypressed_callback` callbacks will be executed even when the event was
+-- consumed by a `keybinding`.
+--
+-- By default, keybindings block those callbacks from being executed.
+--
+-- @property mask_event_callback
+-- @param[opt=true] boolean
+-- @see keybindings
+-- @see keyreleased_callback
+-- @see keypressed_callback
+
+--- Do not call the callbacks on modifier keys (like `Control` or `Mod4`) events.
+-- @property mask_modkeys
+-- @param[opt=false] boolean
+-- @see mask_event_callback
+
+--- Export all keygrabber keybindings as root (global) keybindings.
+--
+-- When this is enabled, calling all of the keygrabber object `keybinding`s will
+-- will create root `awful.key` and will automatically starts the grabbing.
+--
+-- Use this with caution. If many keygrabber or "real" root keybindings are set
+-- on the same key combination, they are all executed and there is almost no
+-- safe way to undo that. Also, while `is_enabled` can be used to temporarely
+-- turn off this behavior, it currently cannot be detached completely. Make sure
+-- the `keygrabber` that use this option have a single instance.
+--
+-- @property export_keybindings
+-- @param[opt=false] boolean
+
+--- The root (global) keybinding to start this keygrabber.
+--
+-- Instead of adding an entry to `root.keys` or `rc.lua` `globalkeys` section,
+-- this property can be used to take care of everything. This way, it becomes
+-- easier to package modules using keygrabbers.
```

Comment:
Hm... so that's your use case? Hm...
Hm...

Hm...
Doesn't this basically mean that you encourage packages with poor customisability? Users might not ever see the keybindings that are added. The packages that I saw so far expire functions to call in the right places, I think, which makes them more complicated to set up, true, but also makes it clearer what is going on.
If I remember correctly, I saw instructions like "add this to your global keybindings definitions". Is that too hard for users?

Hm... :-/
I'm not sure what to think about this.